src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,140 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.commands.AddDeadlineCommand;
+import duke.commands.AddEventCommand;
+import duke.commands.AddTodoCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.DoneCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+
+/**
+ * Represents the Parser class which makes sense of what the user typed.
+ */
+public class Parser {
+
+    /**
+     * Method to parse and make sense what the user typed in.
+     *
+     * @param commandLine the user's input.
+     * @param tasks       the TaskList that contains all the user's tasks.
+     * @return Command command deciphered from the user input.
+     * @throws DukeException If the user gives a bad input.
+     */
+    public Command getCommand(String commandLine, TaskList tasks) throws DukeException {
+        String[] fullCommand = commandLine.split(" ");
+        String command = fullCommand[0];
+        String desc;
+
+        switch (command) {
+        case "bye":
+            return new ExitCommand("exit");
+        case "list":
+            return new ListCommand("list");
+        case "done":
+            if (fullCommand.length == 1) {
+                throw new DukeException("Please give an index number");
+            }
+            int doneIndex = Integer.parseInt(fullCommand[1]);
+            checkIndex(doneIndex, tasks.size());
+            return new DoneCommand("done", doneIndex);
+        case "delete":
+            if (fullCommand.length == 1) {
+                throw new DukeException("Please give an index number");
+            }
+            int deleteIndex = Integer.parseInt(fullCommand[1]);
+            checkIndex(deleteIndex, tasks.size());
+            return new DeleteCommand("delete", deleteIndex);
+        case "todo":
+            desc = commandLine.replace("todo", "").trim();
+            if (desc.equals("")) {
+                throw new DukeException("A to-do needs a description");
+            }
+            return new AddTodoCommand(desc);
+        case "deadline":
+            String by;
```

Comment:
For short variable name declarations, consider adding a small inline comment describing it. Like so:

```java
String by; // deadline for Deadline Task
```